os-client/src/main/kotlin/io/provenance/scope/objectstore/client/CachedOsClient.kt

```diff
@@ -154,18 +167,25 @@ class CachedOsClient(val osClient: OsClient, osDecryptionWorkerThreads: Short, o
         hash: ByteArray,
         encryptionKeyRef: KeyRef,
     ): ListenableFuture<ByteArray> {
+        val span = tracer.buildSpan("GetRawBytes OS").start()
+
         // TODO add good error like we had previously
         val cacheKey = RecordCacheKey(encryptionKeyRef.publicKey, hash.base64String())
         val record = recordCache.asMap()[cacheKey]
 
         return if (record != null) {
-            SettableFuture.create<ByteArray>().also { it.set(record) }
+            SettableFuture.create<ByteArray>().also { it.set(record) }.also {
```

Comment:
This has nothing to do with tracing, but I think we could change these cached cases to return an ImmediateFuture instead of this SettableFuture, might be cleaner. Currently the second `.also` is kind of redundant in this case, could do all 3 statements in one with the current SettableFuture.

If you change to the ImmediateFuture it would look like this 
```
Futures.immediateFuture(record).also {
<tracing stuff>
}
```

Reply:
👍